mlir/include/mlir/Dialect/LLVMIR/ROCDLOps.td

```diff
@@ -204,6 +204,14 @@ def ROCDL_ReadlaneOp : ROCDL_IntrOp<"readlane", [], [0], [AllTypesMatch<["res",
    }];
 }
 
+// the intrinsic function name is too long so we use a shorter name for rocdl.
+def ROCDL_WaveIdOp :  LLVM_IntrOpBase<ROCDL_Dialect, "wave_id",
+                        "amdgcn_s_get_waveid_in_workgroup", [], [], [Pure], 1>,
```

Comment:
Is this really `Pure`? In my mental model, this would have a `read` memory effect on the hw register with the subgroup ID. `Pure` means speculatable + no memory effects

Reply:
I think you are right. this reads hardware status and is not a pure function.

Reply:
Turns out the `Pure` definition is not exactly what we thought.